test/test_iMPOConversions.jl

```diff
@@ -32,6 +61,42 @@ function ITensors.expect(ψ::InfiniteCanonicalMPS, h::InfiniteMPO)
   return expect(ψ, terminate(h)) #defer to src/infinitecanonicalmps.jl
 end
 
+function generate_edges(h::InfiniteMPOMatrix)
+  Ncell = nsites(h)
+  # left termination vector
+  Ls = Matrix{ITensor}(undef, 1, size(h[1], 1))
+  Ls[1] = ITensor(0)
+  for x in 2:(size(h[0], 2) - 1)
+    il0 = commonind(h[0][1, x], h[1][x, 1])
+    Ls[x] = ITensor(0, il0)
+  end
+  Ls[end] = ITensor(1)
+
+  Rs = Vector{ITensor}(undef, size(h[Ncell], 2))
+  Rs[1] = ITensor(1)
+  for x in 2:(size(h[Ncell + 1], 1) - 1)
+    ir0 = commonind(h[Ncell + 1][x, 1], h[Ncell][1, x])
+    Rs[x] = ITensor(0, ir0)
+  end
+  Rs[end] = ITensor(0)
+  return Ls, Rs
+end
+
+function ITensors.expect(ψ::InfiniteCanonicalMPS, h::InfiniteMPOMatrix)
+  Ncell = nsites(h)
+  L, R = generate_edges(h)
+  l = commoninds(ψ.AL[0], ψ.AL[1])
+  L = map(a -> contract(a, δ(l, dag(prime(l)))), L)
+  r = commoninds(ψ.AR[Ncell + 1], ψ.AR[Ncell])
+  R = map(a -> contract(a, δ(r, dag(prime(r)))), R)
+  L = map(a -> contract(a, ψ.C[0], dag(prime(ψ.C[0]))), L)
+  for j in 1:nsites(ψ)
+    temp = map(a -> contract(a, ψ.AR[j], dag(prime(ψ.AR[j]))), h[j])
+    L = L * temp
+  end
+  return (L * R)[1][1]#ITensorInfiniteMPS.scalar_product(L, R)[1]
```

Comment:
Could you remove the trailing comment?

Also, it may be better to use `(L * R)[][]` since then it will check that they are actually scalar values as expected, for example:
```julia
julia> randn(1)[]
-0.5989126360206526

julia> randn(2)[]
ERROR: BoundsError: attempt to access 2-element Vector{Float64} at index []
Stacktrace:
 [1] throw_boundserror(A::Vector{Float64}, I::Tuple{})
   @ Base ./abstractarray.jl:744
 [2] checkbounds
   @ ./abstractarray.jl:709 [inlined]
 [3] _getindex
   @ ./abstractarray.jl:1328 [inlined]
 [4] getindex(::Vector{Float64})
   @ Base ./abstractarray.jl:1296
 [5] top-level scope
   @ REPL[6]:1

julia> randn(1)[1]
-1.1477415235068003

julia> randn(2)[1]
-0.5202834767819509
```